api/src/main/java/org/apache/iceberg/FileScanTask.java

```diff
@@ -29,6 +29,13 @@ public interface FileScanTask extends ContentScanTask<DataFile>, SplittableScanT
    */
   List<DeleteFile> deletes();
 
+  /**
+   * Return the schema for this file scan task.
+   */
+  default Schema schema() {
```

Comment:
this is needed so that `FileScanTaskParser` (added in this PR) can serialize the schema. Then during the deserialization part, schema can be pass into the constructor of `BaseFileScanTask`.

Keep it at this level (not base interface) to limit the scope of change.